shell/platform/linux/fl_renderer.cc

```diff
@@ -314,6 +315,22 @@ void fl_renderer_render(FlRenderer* self, int width, int height) {
 
   g_return_if_fail(FL_IS_RENDERER(self));
 
+  // Save bindings that are set by this function.  All bindings must be restored
+  // to their original values because Skia expects that its bindings have not
+  // been altered.
+  GLint saved_texture_binding;
+  glGetIntegerv(GL_TEXTURE_BINDING_2D, &saved_texture_binding);
+  GLint saved_vao_binding;
+  glGetIntegerv(GL_VERTEX_ARRAY_BINDING, &saved_vao_binding);
+  GLint saved_array_buffer_binding;
+  glGetIntegerv(GL_ARRAY_BUFFER_BINDING, &saved_array_buffer_binding);
+
+  fml::ScopedCleanupClosure restore_bindings([&] {
```

Comment:
I would not use `fml::ScopedCleanupClosure` here - it gets somewhat confusing mixing the C style GObject code with C++ patterns. In this case it doesn't seem dangerous explicitly restoring these at the end of the function as was done in the original commit.

Reply:
Reverted to the original implementation.